pallets/transfer-allowlist/src/lib.rs

```diff
@@ -0,0 +1,481 @@
+		///
+		/// Running this for an existing allowance generates a new allowance based on the current delay, or lack thereof
+		pub fn add_transfer_allowance(
+			origin: OriginFor<T>,
+			currency_id: T::CurrencyId,
+			receiver: Location,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+
+			let allowance_details = match Self::sender_currency_delay(&account_id, currency_id) {
+				Some(delay) => AllowanceDetails {
+					allowed_at: <frame_system::Pallet<T>>::block_number().saturating_add(delay),
+					..AllowanceDetails::default()
+				},
+				_ => AllowanceDetails::default(),
+			};
+			if !<AccountCurrencyTransferAllowance<T>>::contains_key((
+				&account_id,
+				&currency_id,
+				&receiver,
+			)) {
+				Self::increment_or_create_allowance_count(&account_id, &currency_id)?;
+				T::ReserveCurrency::reserve(&account_id, T::Deposit::get())?;
+			};
+			<AccountCurrencyTransferAllowance<T>>::insert(
+				(&account_id, &currency_id, &receiver),
+				&allowance_details,
+			);
+
+			Self::deposit_event(Event::TransferAllowanceCreated {
+				sender_account_id: account_id,
+				currency_id,
+				receiver,
+				allowed_at: allowance_details.allowed_at,
+				blocked_at: allowance_details.blocked_at,
+			});
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(2, 2).ref_time())]
+		/// Restricts a transfer allowance for a sending account/currency/receiver location to:
+		/// - either the current block + delay if a delay is set
+		/// - or the current block if no delay is set
+		pub fn remove_transfer_allowance(
+			origin: OriginFor<T>,
+			currency_id: T::CurrencyId,
+			receiver: Location,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+
+			let blocked_at = match Self::sender_currency_delay(&account_id, currency_id) {
+				Some(delay) => <frame_system::Pallet<T>>::block_number().saturating_add(delay),
+				_ => <frame_system::Pallet<T>>::block_number(),
+			};
+			match <AccountCurrencyTransferAllowance<T>>::get((&account_id, &currency_id, &receiver))
+			{
+				Some(existing_allowance) => {
+					let allowance_details = AllowanceDetails {
+						blocked_at,
+						..existing_allowance
+					};
+					<AccountCurrencyTransferAllowance<T>>::insert(
+						(&account_id, &currency_id, &receiver),
+						&allowance_details,
+					);
+					Self::deposit_event(Event::TransferAllowanceRemoved {
+						sender_account_id: account_id,
+						currency_id,
+						receiver,
+						allowed_at: allowance_details.allowed_at,
+						blocked_at: allowance_details.blocked_at,
+					});
+					Ok(())
+				}
+				None => Err(DispatchError::from(Error::<T>::NoMatchingAllowance)),
+			}
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1, 2).ref_time())]
+		/// Removes a transfer allowance for a sending account/currency and receiving location
+		/// Decrements or removes the sending account/currency count.
```

Comment:
`nit` but I prefer having the doc comments above. I think that's what we do everywhere else

Reply:
👍 fixed